cmd/gaia/cli_test/cli_test.go

```diff
@@ -709,6 +710,53 @@ func TestSlashingGetParams(t *testing.T) {
 	require.NoError(t, err)
 }
 
+func TestGaiaCLIQueryTxPagination(t *testing.T) {
+	t.Parallel()
+	configOverride := cfg.DefaultConfig()
+	configOverride.TxIndex.IndexAllTags = true
+	configOverride.Consensus.SkipTimeoutCommit = true
+	chainID, servAddr, port, gaiadHome, gaiacliHome, _ := initializeFixtures(t, configOverride)
+	flags := fmt.Sprintf("--home=%s --node=%v --chain-id=%v", gaiacliHome, servAddr, chainID)
+
+	// start gaiad git server
+	proc := tests.GoExecuteTWithStdout(t, fmt.Sprintf(
+		"gaiad start --home=%s --rpc.laddr=%v", gaiadHome, servAddr))
+
+	defer proc.Stop(false)
+	tests.WaitForTMStart(port)
+	tests.WaitForNextNBlocksTM(2, port)
+
+	fooAddr, _ := executeGetAddrPK(t, fmt.Sprintf("gaiacli keys show foo --output=json --home=%s", gaiacliHome))
+	barAddr, _ := executeGetAddrPK(t, fmt.Sprintf("gaiacli keys show bar --output=json --home=%s", gaiacliHome))
+
+	for i := 1; i <= 30; i++ {
+		success := executeWrite(t, fmt.Sprintf(
+			"gaiacli tx send %v --amount=%dfooToken --to=%s --from=foo",
+			flags, i, barAddr), app.DefaultKeyPass)
+		require.True(t, success)
+		tests.WaitForNextNBlocksTM(1, port)
+	}
+
+	// perPage = 15, 2 pages
+	txs := executeGetTxs(t, fmt.Sprintf("gaiacli query txs --tags='sender:%s' %v --limit=15 --page=1", fooAddr, flags))
+	require.Len(t, txs, 15)
+	txsPage2 := executeGetTxs(t, fmt.Sprintf("gaiacli query txs --tags='sender:%s' %v --limit=15 --page=2", fooAddr, flags))
+	require.Len(t, txsPage2, 15)
+	txsPage3 := executeGetTxs(t, fmt.Sprintf("gaiacli query txs --tags='sender:%s' %v --limit=15 --page=3", fooAddr, flags))
+	require.Len(t, txsPage3, 15)
+	require.Equal(t, txsPage2, txsPage3)
+
+	// perPage = 16, 2 pages
+	txs = executeGetTxs(t, fmt.Sprintf("gaiacli query txs --tags='sender:%s' %v --limit=16 --page=1", fooAddr, flags))
+	require.Len(t, txs, 16)
+	txs = executeGetTxs(t, fmt.Sprintf("gaiacli query txs --tags='sender:%s' %v --limit=16 --page=2", fooAddr, flags))
+	require.Len(t, txs, 14)
+
+	// perPage = 50
+	txs = executeGetTxs(t, fmt.Sprintf("gaiacli query txs --tags='sender:%s' %v --limit=50 --page=1", fooAddr, flags))
+	require.Len(t, txs, 30)
```

Comment:
can we also concat the previous tx results (16 y 14 results) into a single one and check that this result matches with that concatenated array ?